Core/GDCore/IDE/Events/ArbitraryEventsWorker.cpp

```diff
@@ -165,6 +161,45 @@ void ReadOnlyArbitraryEventsWorker::StopAnyEventIteration() {
   shouldStopIteration = true;
 }
 
+ArbitraryEventsWorkerWithContext::~ArbitraryEventsWorkerWithContext() {}
+
+bool ArbitraryEventsWorkerWithContext::VisitEvent(gd::BaseEvent &event) {
+  // TODO Should CanHaveVariables be used instead?
```

Comment:
I'm not sure why we should use CanHaveVariables. Here we use HasVariables, because we don't want to create a new context if the events has no variables. This make sense like this, no?